src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,34 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(name) {
+    if (name.hidden === false && name instanceof Herbivore) {
+      name.health -= 50;
+    }
+
+    if (name.health === 0) {
+      const index = Animal.alive.indexOf(this);
+
+      Animal.alive = Animal.alive.splice(index, 1);
```

Comment:
Use Animal.alive.splice(index, 1) instead of Animal.alive = Animal.alive.splice(index, 1) to remove the dead animal from the array.